webproject/src/main/java/com/google/sps/servlets/FormHandlerServlet.java

```diff
@@ -69,11 +56,10 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
 
         String[] array = referrer.split("\\?");
         String roomID = array[1];
-        FirebaseDatabase.getInstance()
-            .getReference("messages")
-            .child(roomID)
-            .push()
-            .setValueAsync(new Message(username, imageUrl, "image"));
+
+        String imageJson = gson.toJson(new Message(username, imageUrl, "image"));
+        Firebase.sendRequest("https://summer20-sps-47.firebaseio.com/messages/" + roomID + ".json", "POST", imageJson);
```

Comment:
actually the right way to do it is to use FirebaseDatabase.getInstance().getReference() etc instead of using json
Arjavi is working on it partially in the dependency injection PR. please take a look and continue from there

Reply:
Ok! I'll change the implementation to that way then!